### ISSUE TITLE
SQLFluff fix for oracle package with named parameter

### ISSUE THREAD

**Issue by sureshHARDIYA** [sqlfluff/sqlfluff#4920]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have configured DBT with `sqlfluff` and it works fine. However, we have an oracle package which has some function that takes too many arguments. 

So, we used named arguments as follows:

```sql
select * from table(
  example_function(
    length => 4, 
    breadth => 5
   )
)
```

The code compiles fine, and the linting works okay showing errors in the file. However, when we run `sqlfluff parse PATH` it throws error like:
```
==== parsing violations ====
L:  50 | P:  45 |  PRS | Line 21, Position 45: Found unparsable section: " => '5',\n       
                       | arg..."
```

### Expected Behaviour

It should have fixed SQL indentation on other parts of the file and understood named arguments. 

### Observed Behaviour

Throws error: 

```
==== parsing violations ====
L:  50 | P:  45 |  PRS | Line 21, Position 45: Found unparsable section: " => '5',\n       
                       | arg..."
```

### How to reproduce

Create a SQL that uses a function with named arguments. 

### Dialect

--  oracle 

### Version

2.1.1

### Configuration

.venv/
public/
macros/
target/
dbt_packages/
macros/dbt_utils/

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
